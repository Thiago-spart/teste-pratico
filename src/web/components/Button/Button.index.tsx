import type { ButtonProps } from "./Button.types";

import * as S from "./Button.styled";

export const Button: React.FC<ButtonProps> = ({
	children,
	schema = "primary",
	...rest
}) => {
	return (
		<S.ButtonStyle schema={schema} {...rest}>
			{children}
		</S.ButtonStyle>
	);
};
