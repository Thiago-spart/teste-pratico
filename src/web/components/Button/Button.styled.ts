import styled from "styled-components";

interface ButtonStyleProps {
	schema: "primary" | "secondary";
}

export const ButtonStyle = styled.button<ButtonStyleProps>`
	display: flex;
	align-items: center;
	justify-content: center;
	padding: 1.2rem 1rem;
	font-weight: 300;
	font-size: 1.6rem;
	line-height: 2rem;
	letter-spacing: 0.15rem;
	border-radius: 0.75rem;
	border: none;
	color: #fafafa;
	background: #323232;
	transition: 0.2s;
	cursor: pointer;

	&:focus-visible {
		box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
		outline: none;
	}

	&:hover {
		filter: brightness(0.8);
	}
`;
