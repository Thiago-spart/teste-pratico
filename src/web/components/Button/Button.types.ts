import type { ButtonHTMLAttributes, ReactNode } from "react";

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
	schema: "primary" | "secondary";
	children: ReactNode;
}
