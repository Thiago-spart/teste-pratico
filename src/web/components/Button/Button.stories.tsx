/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { Button } from "./Button.index";

const meta: Meta<typeof Button> = {
	title: "Componentes/Button",
	component: Button,
};

const Template: ComponentStory<typeof Button> = props => <Button {...props} />;

export const Primary = Template.bind({});
Primary.args = {
	children: "Primary",
	type: "button",
};

export default meta;
