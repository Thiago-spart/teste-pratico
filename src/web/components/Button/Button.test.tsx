import { composeStories } from "@storybook/testing-react";
import { fireEvent, render, screen } from "@testing-library/react";

// import userEvent from "@testing-library/user-event";

import * as stories from "./Button.stories";

const { Primary } = composeStories(stories);

describe("Button component", () => {
	it("should match snapshot", () => {
		const { container } = render(<Primary />);

		expect(container).toMatchSnapshot();
	});

	it("should be clicked", () => {
		const onClickMock = jest.fn();
		render(<Primary onClick={onClickMock} />);

		const buttonElement = screen.getByText("Primary");

		fireEvent.click(buttonElement);

		expect(onClickMock).toBeCalled();
	});
});
