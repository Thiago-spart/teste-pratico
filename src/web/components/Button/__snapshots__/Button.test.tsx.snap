// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <button
    class="sc-bczRLJ gjUQPQ"
    type="button"
  >
    Primary
  </button>
</div>
`;
